packages/app/index.ts

```diff
@@ -18,6 +18,23 @@
 import { FirebaseNamespace } from '@firebase/app-types';
 import { createFirebaseNamespace } from './src/firebaseApp';
 
+// Node detection logic from: https://github.com/iliakan/detect-node/
+let isNode = false;
+try {
+  isNode =
+    Object.prototype.toString.call(global.process) === '[object process]';
+} catch (e) {}
+
+isNode &&
+  console.warn(`
+Warning: This is a browser-targeted Firebase bundle but it appears it is being run in a Node
+environment.  This may be because you are using ES module resolution and the Node bundle is
```

Comment:
It's not clear to me what ES module resolution is. It's probably better to just state please use the "main" field in package.json.
We can also add the link to webpack doc - https://webpack.js.org/configuration/resolve/#resolvemainfields
and rollup doc - https://github.com/rollup/rollup-plugin-node-resolve

Reply:
Modified the text.